pkg/clusteragent/admission/mutate/auto_instrumentation.go

```diff
@@ -319,6 +319,13 @@ func injectAutoInstruConfig(pod *corev1.Pod, libsToInject []libInfo) error {
 		}
 	}
 
+	// try to inject all if the annotation is set
+	err := injectLibConfig(pod, "all")
+	if err != nil {
```

Comment:
I think the error management is not clear. On line 331, we return `lastError` but if we have an error here, we don't update `lastError` is it expected? 

also we can reduce the scope the `err` here

```suggestion
	
	if err := injectLibConfig(pod, "all"); err != nil {
	    lastError = err
```

Reply:
Good catch! Fixed and added some erroneous test cases